hello_network/lib/hello_network.ex

```diff
@@ -1,18 +1,62 @@
 defmodule HelloNetwork do
+  @moduledoc """
+  Example of setting up Networking on a Nerves device.
+  """
 
-  alias Nerves.Networking
+  alias Nerves.{Network, Udhcpc}
 
-  @interface :eth0
+  @interface "eth0"
+  @settings [ipv4_address_method: :dhcp]
 
+  @doc "Main entry point into the program. This is an OTP callback."
   def start(_type, _args) do
-    unless :os.type == {:unix, :darwin} do     # don't start networking unless we're on nerves
-      {:ok, _} = Networking.setup @interface
-    end
-    {:ok, self()}
+    GenServer.start_link(__MODULE__, {@interface, @settings}, [name: __MODULE__])
   end
 
-  @doc "Attempts to perform a DNS lookup to test connectivity."
+  @doc "Are we connected to the internet?"
+  def connected?, do: GenServer.call(__MODULE__, :connected?)
+
+  @doc "Returns the current ip address"
+  def ip_addr, do: GenServer.call(__MODULE__, :ip_addr)
+
+  @doc """
+  Attempts to perform a DNS lookup to test connectivity.
+
+  ## Examples
+
+    iex> HelloNetwork.test_dns()
+    {:ok,
+     {:hostent, 'nerves-project.org', [], :inet, 4,
+      [{192, 30, 252, 154}, {192, 30, 252, 153}]}}
+  """
   def test_dns(hostname \\ 'nerves-project.org') do
     :inet_res.gethostbyname(hostname)
   end
+
+  ## GenServer callbacks
+
+  def init({interface, settings}) do
+    Network.setup(interface, settings)
+    {:ok, _} = Registry.register(Udhcpc, interface, [])
```

Comment:
We _could_ hook into the `Udhcpc` Registry today, but, the intention is that all underlying connections to Registry will be replaced with SystemRegistry. Using SystemRegistry will ensure that there is no race condition between the time that we register and the time that we get the change. 

Here is an example that uses system registry to listen for the ip_address changes
https://github.com/mobileoverlord/nerves_init_net_kernel/blob/master/lib/nerves_init_net_kernel/manager.ex